spec/services/idv/financials_validator_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe Idv::FinancialsValidator do
+  let(:user) { build(:user) }
+
+  let(:idv_session) do
+    idvs = Idv::Session.new({}, user)
+    idvs.vendor = :mock
+    idvs.resolution = Proofer::Resolution.new session_id: 'some-id'
+    idvs
+  end
+
+  let(:session_id) { idv_session.resolution.session_id }
+
+  let(:params) do
+    { ccn: '123-45-6789' }
+  end
+
+  let(:confirmation) { instance_double(Proofer::Confirmation) }
+
+  subject { Idv::FinancialsValidator.new(idv_session: idv_session, vendor_params: params) }
+
+  def stub_agent_calls
+    agent = instance_double(Idv::Agent)
+    allow(Idv::Agent).to receive(:new).
+      with(applicant: idv_session.applicant, vendor: :mock).
+      and_return(agent)
+    expect(agent).to receive(:submit_financials).
+      with(params, idv_session.resolution.session_id).and_return(confirmation)
+  end
+
+  describe '#success?' do
+    it 'returns Proofer::Confirmation#success?' do
+      stub_agent_calls
+
+      expect(confirmation).to receive(:success?).and_return('yay!')
+
+      expect(subject.success?).to eq 'yay!'
```

Comment:
seeing `yay` in the expectation kind of threw me here as I wasn't sure where it was coming from. 

Assign string to a variable in the spec setup and use that variable throughout the test instead?

Same for spec below (`foobar`)